lib/scim_rails/encoder.rb

```diff
@@ -0,0 +1,25 @@
+require "jwt"
+
+module ScimRails
+  module Encoder
+    extend self
+
+    def encode(company)
+      payload = {
+        iat: Time.current.to_i,
+        ScimRails.config.basic_auth_model_searchable_attribute =>
+          company.public_send(ScimRails.config.basic_auth_model_searchable_attribute)
+      }
+
+      JWT.encode(payload, ScimRails.config.signing_secret, ScimRails.config.signing_algorithm)
```

Comment:
If we don't provide a signing secret and algorithm (the defaults) would it be possible for anyone to create a token if they know someone's basic_auth_model_searchable_attribute. Is that what the IAT was for?

Reply:
Yes. Without signing configuration, one could generate the exact token if they know someone's basic_auth_model_searchable_attribute just because of the way JWT works. IAT adds some randomness.

It is highly advised to provide this configuration, but I wanted to support all options JWT supports.

Reply:
I guess we could force people to set it by throwing an error in the `encode/decode` methods. What do you think @wernull

Reply:
No, I think it's ok to leave the defaults empty for now. With the IAT it would take a lot of brute forcing to figure it out and ideally people would either have monitoring or throttling to handle a brute force in authentication